intel_adsp: select log_backend_xtensa_sim for simulator

laurenmurphyx64 opened the pull request:
Workaround for logging test failures on intel_adsp_ace*/sim first observed in [weekly build](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18811754455).

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/98451

lyakh commented on soc/intel/intel_adsp/ace/Kconfig.defconfig.series:
```diff
@@ -52,8 +52,15 @@ config DYNAMIC_INTERRUPTS
 	default y
+endif
+
+if SIMULATOR_XTENSA
```
does Kconfig have an `else`?

laurenmurphyx64 committed:
intel_adsp: select log_backend_xtensa_sim for simulator

ADSP logging backend requires winstream console,
which is turned off for simulator.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>
